cognee/api/health.py

```diff
@@ -0,0 +1,327 @@
+"""Health check system for cognee API."""
+
+import time
+import asyncio
+from datetime import datetime, timezone
+from typing import Dict, Any, Optional
+from enum import Enum
+from pydantic import BaseModel
+
+from cognee.version import get_cognee_version
+from cognee.shared.logging_utils import get_logger
+
+logger = get_logger()
+
+
+class HealthStatus(str, Enum):
+    HEALTHY = "healthy"
+    DEGRADED = "degraded"
+    UNHEALTHY = "unhealthy"
+
+
+class ComponentHealth(BaseModel):
+    status: HealthStatus
+    provider: str
+    response_time_ms: int
+    details: str
+
+
+class HealthResponse(BaseModel):
+    status: HealthStatus
+    timestamp: str
+    version: str
+    uptime: int
+    components: Dict[str, ComponentHealth]
+
+
+class HealthChecker:
+    def __init__(self):
+        self.start_time = time.time()
+
+    async def check_relational_db(self) -> ComponentHealth:
+        """Check relational database health."""
+        start_time = time.time()
+        try:
+            from cognee.infrastructure.databases.relational.get_relational_engine import (
+                get_relational_engine,
+            )
+            from cognee.infrastructure.databases.relational.config import get_relational_config
+
+            config = get_relational_config()
+            engine = get_relational_engine()
+
+            # Test connection by creating a session
+            session = await engine.get_session()
+            if session:
+                await session.close()
+
+            response_time = int((time.time() - start_time) * 1000)
+            return ComponentHealth(
+                status=HealthStatus.HEALTHY,
+                provider=config.db_provider,
+                response_time_ms=response_time,
+                details="Connection successful",
+            )
+        except Exception as e:
+            response_time = int((time.time() - start_time) * 1000)
+            return ComponentHealth(
```

Comment:
Let's log the stacktrace, as we're currently missing out on it if anything breaks

```suggestion
            response_time = int((time.time() - start_time) * 1000)
            logger.error(f"Relational DB health check failed: {str(e)}", exc_info=True)
            return ComponentHealth(
```